### ISSUE TITLE
useLocalStorage set function does not entirely comply with useState

### ISSUE THREAD

**Issue by valyrie97** [juliencrn/usehooks-ts#204]
the set function given by useState, guarantees that if it is called with a function argument, the first parameter will be the current value of the state. However, in useLocalStorage, this is not always true. in the case that a render does not happen between two set calls.

Example:

```javascript
const [n, setN] = useState(0);

useEffect(() => {
  setN(n => {
    console.log('state value:', n);
    return n + 1;
  });
  setN(n => {
    console.log('state value:', n);
    return n + 1;
  });
}, []);
```

This code will print out
```
0
1
```

useLocalStorage on the other hand:

```javascript
const [n, setN] = useLocalStorage('example-key', 0);

useEffect(() => {
  setN(n => {
    console.log('state value:', n);
    return n + 1;
  });
  setN(n => {
    console.log('state value:', n);
    return n + 1;
  });
}, [])
```
```
0
0
```

The problem stems from [line 55](https://github.com/juliencrn/usehooks-ts/blob/2dc20df615b32430b943349e8833b74a1fbc33dc/lib/src/useLocalStorage/useLocalStorage.ts#L55) where the stored value is used instead of reading the current value from localstorage. I copied the hook locally and changed it to `readValue()` and this fixed the issue.

Im sure theres something to be said here of performance, but i havent done any testing. possibly using a ref would be better suited for the task. Id be happy to whip up an implementation if this is something you're interested in!

Edit: off by one errors in code

**Comment by juliencrn** [comment]
Hi @valyrie97, thanks for reporting. Yes, if you want to create a PR, it's welcome, else, could you share your current implementation 🙏

**Comment by justinmetros** [comment]
Just ran into this. It affects useDarkMode as well, since useLocalStorage is under the hood of that. 

Fix in PR #242  corrects the issue for me.

**Comment by thenick775** [comment]
Bump, same here, pr 242 fixes for me as well.

**Comment by juliencrn** [comment]
Fix from @valyrie97 's #242 applied on #436, it will be in the next release this week, thanks all 👍
